toptal/picasso
# fix(DatePicker): fix value reset and activeMonth of ranged DatePicker

[FX-1836]

### Description

There are two bugs in the DatePicker (in Range Mode) component that I found.
1. `Invalid time value` error is raised after re-render with new value
2. Input doesn't allow to remove its value by selecting text and deleting it with the backspace/delete button

### How to reproduce
* Open https://picasso.toptal.net/?path=/story/picasso-lab-datepicker--datepicker#range-mode
* Select a date range, select all text inside the input and click the backspace key
* The input value remains the same instead of becoming empty.
* Edit the code of the range component by adding `hideOnSelect={false}` to the `DatePicker` component
* After the component is updated, try to select a date range
* You will see `RangeError: Invalid time value` at the bottom of the component container.

### How to test

- Open the demo https://picasso.toptal.net/fx-1836-fix-datepicker-reset/?path=/story/picasso-lab-datepicker--datepicker#range-mode
- Repeat steps from the `How to reproduce` section

### Review


<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-1836]: https://toptal-core.atlassian.net/browse/FX-1836?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
